src/rust/engine/process_execution/src/docker_tests.rs

```diff
@@ -541,6 +541,38 @@ async fn all_containing_directories_for_outputs_are_created() {
   assert_eq!(result.original.platform, platform_for_tests().unwrap());
 }
 
+#[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+async fn outputs_readable_only_by_container_user_are_captured() {
+  skip_if_no_docker_available_in_macos_ci!();
+
+  let result = run_command_via_docker(
+    Process::new(vec![
+      SH_PATH.to_string(),
+      "-c".to_owned(),
+      format!(
+        // Ensure that files are only readable by the container user (which on Linux would usually
+        // mean that a non-root user outside the container would not have access).
+        "/bin/mkdir birds/falcons && echo -n {} > cats/roland.ext && chmod o-r -R birds cats",
+        TestData::roland().string()
+      ),
+    ])
+    .output_files(relative_paths(&["cats/roland.ext"]).collect())
+    .output_directories(relative_paths(&["birds/falcons"]).collect())
+    .docker(IMAGE.to_owned()),
+  )
+  .await
+  .unwrap();
+
+  assert_eq!(result.stdout_bytes, "".as_bytes());
+  assert_eq!(result.stderr_bytes, "".as_bytes());
+  assert_eq!(result.original.exit_code, 0);
+  assert_eq!(
```

Comment:
Is the process sandbox able to be cleaned up on the host side after reading the outputs or does it silently leak due to insufficient perms to recursively remove the sandbox on the host?

Reply:
The latter: see https://github.com/pantsbuild/pants/pull/18327#discussion_r1114560501